modules/sdk/src/main/scala/org/tessellation/sdk/domain/healthcheck/services/HealthCheckConsensus.scala

```diff
@@ -1,32 +1,65 @@
 package org.tessellation.sdk.domain.healthcheck.services
 
-import cats.effect.Ref
+import cats.Applicative
+import cats.effect.{Ref, Spawn}
 import cats.syntax.bifunctor._
 import cats.syntax.flatMap._
 import cats.syntax.functor._
 import cats.syntax.traverse._
-import cats.{Applicative, Monad}
 
-import org.tessellation.sdk.domain.healthcheck.types.ConsensusRounds
+import org.tessellation.schema.peer.PeerId
+import org.tessellation.sdk.domain.healthcheck.types._
 
-abstract class HealthCheckConsensus[F[_]: Monad]() extends HealthCheck[F] {
+abstract class HealthCheckConsensus[F[_]: Spawn]() extends HealthCheck[F] {
   def allRounds: Ref[F, ConsensusRounds[F]]
   def roundsInProgress = allRounds.get.map(_.inProgress)
+  def historicalRounds = allRounds.get.map(_.historical)
+
+  def peersUnderConsensus: F[Set[PeerId]] =
+    roundsInProgress.map(_.keySet.map(_.id))
 
   final override def trigger(): F[Unit] =
     triggerRound()
 
-  def triggerRound(): F[Unit] =
+  private def triggerRound(): F[Unit] =
     roundsInProgress.flatMap { inProgress =>
       if (inProgress.nonEmpty) manageRounds(inProgress) else Applicative[F].unit
     }
 
-  def manageRounds(rounds: ConsensusRounds.InProgress[F]): F[Unit] =
+  private def manageRounds(rounds: ConsensusRounds.InProgress[F]): F[Unit] = {
+    def checkRounds(inProgress: ConsensusRounds.InProgress[F]): F[ConsensusRounds.Finished[F]] =
+      // Note: more logic in next PRs
+      partition(inProgress).map { case (finished, _) => finished }
+
     partition(rounds).flatMap {
-      case (finished, inProgress) => Applicative[F].unit
+      case (finished, inProgress) => checkRounds(inProgress).map(r => (finished ++ r, inProgress -- r.keySet))
+    }.flatMap {
+      case (ready, toManage) =>
+        toManage.values.toList
+          .map(_.manage)
+          .map(Spawn[F].start)
+          .sequence
+          .as(ready)
+    }.flatMap { calculateOutcome }.map {
+      _.filter { case (_, (decision, _)) => decision.isInstanceOf[PositiveOutcome] }
```

Comment:
I guess the logic for `NegativeOutcome` will come in the next PRs?

Reply:
I wonder which one is neater:
`_.filter { case (_, (decision, _)) => decision.isInstanceOf[PositiveOutcome] }`
vs
`_.collect { case outcome @ (_, (PositiveOutcome(), _)) => outcome }`

Reply:
PositiveOutcome will have many arguments, so IMO the first one anyway

Reply:
It can be like this as well:
`_.collect { case outcome @ (_, (_: PositiveOutcome), _)) => outcome }`
but whatever you prefer is ok.